src/index.ts

```diff
@@ -1414,6 +1409,12 @@ export abstract class PinejsClientCore<PinejsClient> {
 			method,
 		};
 
+		if (retry === false || (retry == null && this.retry === false)) {
+			// return the result directly so that we can preserve
+			// the potentially enhanced promise-like result.
+			return this._request(opts);
+		}
+
 		return this.callWithRetry(async () => {
 			return await this._request(opts);
 		}, retry);
```

Comment:
Would it be better to change this to
```suggestion
		return this.callWithRetry(() => this._request(opts), retry);
```
and have pinejs-client-supertest do
```ts
	protected callWithRetry<T>(
		fnCall: () => Promise<T>,
		retry?: RetryParameters,
	): Promise<T> {
		if (retry === false || (retry == null && this.retry === false)) {
			return fnCall();
		}
		throw new Error('Cannot use retry with supertest');
	}
```
?

Reply:
I will try that :+1:

Reply:
Indeed it works :+1: